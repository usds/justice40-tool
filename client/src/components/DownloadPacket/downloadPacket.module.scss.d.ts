declare namespace DownloadPacketModuleScssNamespace {
    export interface IDownloadPacketModuleScss {
        downloadBoxContainer: string;
        downloadBox: string;
        downloadBoxTextBox: string;
        downloadBoxTitle: string;
        downloadBoxText: string;
        downloadBoxButtonContainer: string;
        downloadBoxButton: string;
        downloadPacketText: string;
    }
  }

declare const DownloadPacketModuleScssModule: DownloadPacketModuleScssNamespace.IDownloadPacketModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: DownloadPacketModuleScssNamespace.IDownloadPacketModuleScss;
  };

export = DownloadPacketModuleScssModule;
