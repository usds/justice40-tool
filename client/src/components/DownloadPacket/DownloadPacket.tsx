import React, {FC} from 'react';
import {Button, Grid, Tag} from '@trussworks/react-uswds';

import * as styles from './downloadPacket.module.scss';
import * as METHODOLOGY_COPY from '../../data/copy/methodology';

// @ts-ignore
import downloadIcon from '/node_modules/uswds/dist/img/usa-icons/file_download.svg';

const DownloadPacket = () => {
  // inline components to make build layout below more readable
  const UpdateTag = () => <div className={styles.tagContainer}><Tag
    className={styles.updateTag}>{METHODOLOGY_COPY.DOWNLOAD_PACKAGE.UPDATED_TAG}</Tag></div>;

  const NewTag = () => <div className={styles.tagContainer}>
    <Tag className={styles.newTag}>{METHODOLOGY_COPY.DOWNLOAD_PACKAGE.NEW_TAG}</Tag></div>;

  const DownloadButton: FC = ({children}) =>
    <Button className={styles.downloadBoxButton} type="button">
      <span className={styles.downloadButtonIconSpan}>
        <img src={downloadIcon} className={styles.downloadButtonIcon}
          alt={'download icon for download package'} />
      </span>
      <span className={styles.downloadButtonText} >
        {children}
      </span>
    </Button>;

  return (
    <Grid>
      <div className={styles.downloadBoxContainer}>
        <div className={styles.downloadBox}>
          <div className={styles.downloadBoxTextBox}>

            {/* Download box title */}
            <div className={styles.downloadBoxTitle}>
              {METHODOLOGY_COPY.DOWNLOAD_PACKAGE.TITLE}
            </div>

            {/* Download box description 1 */}
            <div className={styles.downloadSourceText}>
              {METHODOLOGY_COPY.DOWNLOAD_PACKAGE.DESCRIPTION1}
            </div>

            {/* Download box button 1 */}
            <div className={styles.downloadButtonContainer}>
              <UpdateTag/>
              <a data-cy={'download-link'}
                download
                href={METHODOLOGY_COPY.DOWNLOAD_ZIP_URL}>
                <DownloadButton>{METHODOLOGY_COPY.DOWNLOAD_PACKAGE.BUTTON_TEXT1}</DownloadButton>
              </a>
            </div>
            <div className={styles.lastUpdated}>
              {METHODOLOGY_COPY.DOWNLOAD_PACKAGE.ZIP_LAST_UPDATED}
            </div>


            {/* Download box button 2 */}
            <div className={styles.downloadButtonContainer}>
              <UpdateTag/>
              <a data-cy={'shapefile-link'}
                download
                href={METHODOLOGY_COPY.DOWNLOAD_SHAPEFILE_URL}>
                <DownloadButton>{METHODOLOGY_COPY.DOWNLOAD_PACKAGE.BUTTON_TEXT2}</DownloadButton>
              </a>
            </div>
            <div className={styles.lastUpdated}>
              {METHODOLOGY_COPY.DOWNLOAD_PACKAGE.SHAPE_LAST_UPDATED}
            </div>

            {/* Download box button 3 */}
            <div className={styles.downloadButtonContainer}>
              <NewTag/>
              {/* target and rel required since PDFs open in browser and don't download */}
              <a data-cy={'tsd-link'}
                download
                target={'_blank'}
                rel={'noreferrer'}
                href={METHODOLOGY_COPY.DOWNLOAD_TSD_URL}>
                <DownloadButton>{METHODOLOGY_COPY.DOWNLOAD_PACKAGE.BUTTON_TEXT3}</DownloadButton>
              </a>
            </div>
            <div className={styles.lastUpdated}>
              {METHODOLOGY_COPY.DOWNLOAD_PACKAGE.TSD_LAST_UPDATED}
            </div>
          </div>
        </div>
      </div>
    </Grid>
  );
};

export default DownloadPacket;
