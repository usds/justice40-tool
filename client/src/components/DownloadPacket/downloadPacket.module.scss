@use '../../styles/design-system.scss' as *;

.downloadBoxContainer {

    color: whitesmoke;
    margin: auto;
    @include u-margin-top(2.5);

    .downloadBox {
      @include u-bg('blue-80v');
      border-radius: 6px 6px;
      // @include u-margin-top(3);

      .downloadBoxTextBox {
        padding: 25px 25px;
        display: flex;
        flex-direction: column;
        
        .downloadBoxTitle {
          @include typeset('sans', 'xs', 3);
          @include u-text('semibold');
          @include u-margin-bottom(2);
        }
        
        .downloadBoxText {
          @include typeset('sans', 'xs', 3);
          @include u-margin-bottom(4);
          
          span {
            font-style: italic;
          }
        }

        .downloadBoxButtonContainer {
          margin-top: 20px 0 0 0;
          align-self: center;

          a {
            text-decoration: none;
          }

          img {
            // This should be a global css-filter value as it's generated
            // from the primary color.
            filter: invert(15%) sepia(9%) saturate(5666%) hue-rotate(175deg) brightness(96%) contrast(95%);
          }
        }

        .downloadBoxButton{
          @include u-bg('white');
          @include u-color('blue-80v');
          display: flex;

          .downloadPacketText {
            padding-top: 4px;
            padding-left: 5px;
          }
        }

      }
    }
  }