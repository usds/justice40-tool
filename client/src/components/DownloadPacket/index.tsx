import React from 'react';
import {Button, Grid} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';
import {defineMessages} from 'react-intl';

import * as styles from './downloadPacket.module.scss';
import * as constants from '../../data/constants';
// @ts-ignore
import downloadIcon from '/node_modules/uswds/dist/img/usa-icons/file_download.svg';

const DownloadPacket = () => {
  const intl = useIntl();
  const messages = defineMessages({
    downloadPacketHeader: {
      id: 'downloadPacket.header.text',
      defaultMessage: 'Draft communities list (pre-decisional, 137MB)',
      description: 'download packet header text',
    },
    downloadPacketInfo: {
      id: 'downloadPacket.info.text',
      defaultMessage: 'The package includes the draft list of prioritized communities (.csv and .xlsx) and'+
      ' information about how to use the list (.pdf). This information should not be used' +
      ' to make program resource allocation decisions.',
      description: 'download packet info text',
    },
    downloadPacketButtonText: {
      id: 'downloadPacket.button.text',
      defaultMessage: 'Download packet',
      description: 'download packet button text',
    },
  });

  return (
    <Grid>
      <div className={styles.downloadBoxContainer}>
        <div className={styles.downloadBox}>
          <div className={styles.downloadBoxTextBox}>
            <div className={styles.downloadBoxTitle}>{intl.formatMessage(messages.downloadPacketHeader)}</div>
            <div className={styles.downloadBoxText}>
              {intl.formatMessage(messages.downloadPacketInfo)}
            </div>
            <div className={styles.downloadBoxButtonContainer}>
              <a data-cy={'download-link'} href={constants.DOWNLOAD_ZIP_URL}>
                <Button className={styles.downloadBoxButton} type="button">
                  <div><img src={downloadIcon} /> </div>
                  <div className={styles.downloadPacketText}>
                    {intl.formatMessage(messages.downloadPacketButtonText)}
                  </div>
                </Button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </Grid>
  );
};

export default DownloadPacket;
