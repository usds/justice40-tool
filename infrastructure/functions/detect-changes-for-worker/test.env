REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1