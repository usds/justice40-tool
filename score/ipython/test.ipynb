{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
