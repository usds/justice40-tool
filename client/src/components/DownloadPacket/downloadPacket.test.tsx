import * as React from 'react';
import {render, screen} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import DownloadPacket from '.';

test('download packet component defined', () => {
  render(
      <LocalizedComponent>
        <DownloadPacket />
      </LocalizedComponent>,
  );

  screen.getByRole('button', {name: /download packet/i});
});

