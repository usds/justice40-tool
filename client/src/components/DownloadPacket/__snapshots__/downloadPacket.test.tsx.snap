// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`download packet component defined checks if component renders 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="grid"
  >
    <div>
      <div>
        <div>
          <div>
            <span>
              NEW
            </span>
             files available for download
          </div>
          <div>
            Download the data sources used in the CEJST (.csv, .xlxs, 
            <span>
              .pdf
            </span>
             that describes how to use the list, and a codebook, 53MB unzipped), the shapefile, along with a 
            <span>
              codebook
            </span>
             (.zip, 742MB unzipped) or the technical support document (.pdf, 2.5MB unzipped).
          </div>
          <div>
            <div>
              <span
                class="usa-tag"
                data-testid="tag"
              >
                <strong>
                  Updated
                </strong>
              </span>
            </div>
            <a
              data-cy="download-link"
              download=""
              href="//"
            >
              <button
                class="usa-button"
                data-testid="button"
                type="button"
              >
                <span>
                  <img
                    alt="download icon for download package"
                    src="test-file-stub"
                  />
                </span>
                <span>
                  Download data sources
                </span>
              </button>
            </a>
          </div>
          <div>
            Last updated: 04/26/22 
          </div>
          <div>
            <div>
              <span
                class="usa-tag"
                data-testid="tag"
              >
                <strong>
                  Updated
                </strong>
              </span>
            </div>
            <a
              data-cy="shapefile-link"
              download=""
              href="//"
            >
              <button
                class="usa-button"
                data-testid="button"
                type="button"
              >
                <span>
                  <img
                    alt="download icon for download package"
                    src="test-file-stub"
                  />
                </span>
                <span>
                  Download shapefile
                </span>
              </button>
            </a>
          </div>
          <div>
            Last updated: 04/26/22 
          </div>
          <div>
            <div>
              <span
                class="usa-tag"
                data-testid="tag"
              >
                <strong>
                  NEW
                </strong>
              </span>
            </div>
            <a
              data-cy="tsd-link"
              download=""
              href="//"
              rel="noreferrer"
              target="_blank"
            >
              <button
                class="usa-button"
                data-testid="button"
                type="button"
              >
                <span>
                  <img
                    alt="download icon for download package"
                    src="test-file-stub"
                  />
                </span>
                <span>
                  Download technical support document
                </span>
              </button>
            </a>
          </div>
          <div>
            Last updated: 04/19/22 
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
